linux_boot_params/src/lib.rs

```diff
@@ -741,6 +757,10 @@ impl BootParams {
             unsafe { CStr::from_ptr(self.hdr.cmd_line_ptr as *const c_char) }
         }
     }
+
+    pub fn ramdisk(&self) -> Option<Ramdisk> {
+        self.hdr.ramdisk()
+    }
```

Comment:
I'm personally happy with keeping a Ramdisk struct in this crate, and providing a function for getting the ramdisk from the BootParams here. This way we ensure that all create uses the same values.

However, this struct isn't part of the linux boot protocol, so highlighting for review.